src/Carting.Api/Controllers/V1/CartingController.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Carting.Api.Mappers.V1;
+using Carting.Api.Requests.V1;
+using Carting.Core.Services;
+using Carting.DataAccess.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Carting.Api.Controllers.V1;
+
+[ApiController]
+[ApiVersion("1.0")]
+[ApiExplorerSettings(GroupName = "v1")]
+[Route("api/v{version:apiVersion}")]
+public class CartingController : ControllerBase
+{
+    private readonly ICartingService cartingService;
+
+    public CartingController(ICartingService cartingService)
+    {
+        this.cartingService = cartingService;
+    }
+
+    /// <summary>
+    /// Returns cart items for the provided cartId.
+    /// </summary>
+    /// <param name="cartId">Identifier of the cart.</param>
+    /// <returns>Cart items for the provided cartId.</returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     GET /cart/1
+    ///
+    /// </remarks>
+    /// <response code="200">Cart items returned successfully</response>
+    /// <response code="500">Internal server error occured</response>
+    [HttpGet]
+    [Route("cart/{cartId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public IActionResult Get(string cartId)
+    {
+        try
+        {
+            var result = cartingService.GetCartItems(cartId);
+
+            return Ok(Mapper.Map(cartId, result));

```

Comment:
Static mapper, testability is hearting here, readability also, as there's no clear understanding of what is mapped here without going and reading the whole three layers

Reply:
Renamed to CartItemMapper.
I think a mapper can be static, it can be tested individually, and as it hasn't got any dependencies, there is no need to mock it in the classes where it is used. But yes maybe for extensibility an interface/class can be used instead.